src/components/templates/TemplateSearch.jsx

```diff
@@ -1,58 +1,81 @@
+  const [currentPage, setCurrentPage] = useState(1)
 
   const error = useSelector(state => state.selectorReducer.templateSearch.error)
+  const totalResults = useSelector(state => state.selectorReducer.templateSearch.totalResults)
 
-  const search = (query) => {
+  useEffect(() => {
     // Cancel all current searches
     while (tokens.current.length > 0) {
       tokens.current.pop().cancel = true
     }
 
+    const offset = (currentPage - 1) * Config.templateSearchResultsPerPage
     // Create a token for this set of searches
     const token = { cancel: false }
     tokens.current.push(token)
-
-    getTemplateSearchResults(query).then((response) => {
+    getTemplateSearchResults(query, offset).then((response) => {
       if (!token.cancel) dispatch(setTemplateSearchResults(response.results, response.totalHits, response.error))
     })
+  }, [query, currentPage])
+
+  const hasNextButton = totalResults > currentPage * Config.templateSearchResultsPerPage
+
+  const incrementPage = () => {
+    setCurrentPage(currentPage + 1)
+  }
+
+  const updateSearch = (e) => {
+    setCurrentPage(1)
+    setQueryText(e.target.value)
   }
 
   return (
-    <div id="search">
-      <Alert text={error} />
-      <div className="row">
-        <div className="col">
-          <form className="form-inline" onSubmit={event => event.preventDefault()}>
-            <div className="form-group" style={{ paddingBottom: '10px', paddingTop: '10px' }}>
-              <label className="font-weight-bold" htmlFor="searchInput">Find a resource template</label>&nbsp;
-              <div className="input-group" style={{ width: '750px', paddingLeft: '5px' }}>
-                <input id="searchInput" type="text" className="form-control"
-                       onChange={ event => search(event.target.value) }
-                       placeholder="Enter id, label, URI, remark, or author" />
+    <React.Fragment>
+      <div id="search">
+        <Alert text={error} />
+        <div className="row">
+          <div className="col">
+            <form className="form-inline" onSubmit={event => event.preventDefault()}>
+              <div className="form-group" style={{ paddingBottom: '10px', paddingTop: '10px' }}>
+                <label className="font-weight-bold" htmlFor="searchInput">Find a resource template</label>&nbsp;
+                <div className="input-group" style={{ width: '750px', paddingLeft: '5px' }}>
+                  <input id="searchInput" type="text" className="form-control"
+                         onChange={ updateSearch }
+                         placeholder="Enter id, label, URI, remark, or author" />
+                </div>
               </div>
-            </div>
-          </form>
+            </form>
+          </div>
         </div>
       </div>
-    </div>
+
+      <SinopiaResourceTemplates history={props.history} />
+      { hasNextButton && (<button className="btn btn-outline-secondary" onClick={incrementPage}>Next Page</button>)}
```

Comment:
As noted in the ticket, please add pagination similar to Sinopia search results (and re-using the Search Results Paging component).

Reply:
That componetn is not reusable as it is tightly bound to the redux state of the sinopia search.

Reply:
If I'm not mistaken, the structure of `state.selectorReducer.search` and `state.selectorReducer.templateSearch` are similar. It would just need to be unbound such that the redux path of the search result to use needs to be passed in.

Reply:
I think it's a lot more than that.  I'm getting:
<img width="813" alt="Screen Shot 2019-11-25 at 12 01 25 PM" src="https://user-images.githubusercontent.com/92044/69565826-5d5d2180-0f7b-11ea-903f-f3f08e26083d.png">

Reply:
Also encountered this in some refactoring of search. Will fix.